src/core/libraries/kernel/net.cpp

```diff
@@ -0,0 +1,50 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <common/singleton.h>
+#include <core/libraries/libs.h>
+#include "net.h"
+
+namespace Libraries::Kernel {
+int PS4_SYSV_ABI posix_socket(int domain, int type, int protocol) {
+    auto* socket = Common::Singleton<NetPosixInternal>::Instance();
+    return socket->create_socket(domain, type, protocol);
+}
+int PS4_SYSV_ABI posix_connect(int sockfd, const struct sockaddr* addr, socklen_t addrlen) {
+    return 0;
+}
+u32 PS4_SYSV_ABI posix_htonl(u32 hostlong) {
+    return htonl(hostlong);
```

Comment:
may you add ::, for global scope, like `::htonl`. it shows we use OS functions

Reply:
good point